src/TracerState.h

```diff
@@ -1353,6 +1339,27 @@ class TracerState {
     std::vector<unsigned int> object_count_;
     std::vector<std::pair<lifecycle_t, int>> lifecycle_summary_;
     std::vector<unsigned long int> event_counter_;
+
+    std::vector<AssignmentState> assignment_stack_;
+    
+public: 
+  void push_assignment_stack(SEXP symbol, SEXP env, sexptype_t type, SEXP calling_env) {
```

Comment:
I think its better that we make this API object oriented. 
There should be a get_assignment_stack() method. 
We can invoke `peek`, `is_empty`, `pop` and `push` on the stack.